### ISSUE TITLE
List<> of List<> collection initializer fails to populate the list if another project has an internal generic class named List under some circumstances

### ISSUE THREAD

**Issue by jbeshir** [bridgedotnet/Bridge#3186]
If you create an internal generic class named List in one project and namespace, then include a using statement for that namespace in a file in another project, and that file also references System.Collections.Generic and attempts to instantiate a System.Collections.Generic.List<> of a System.Collections.Generic.List<> using a collection initializer, then the collection initializer's corresponding JS will be generated incorrectly and fail to populate the collection.

This also only happens so long as a method instantiating that internal generic class is invoked by code outside its project, even if that code is itself not used.

This was originally a rather mysterious bug in collection initialisers not working in a large project but working every time I tried to replicate outside of the project which got narrowed down by iteratively removing code.

It looks like some kind of fault in type resolution getting confused by C# internal classes.

### Steps To Reproduce

Unfortunately can't be replicated on deck.net due to requiring multiple projects.

You have to create two Bridge.Net projects in the same solution- a main project and a library project that the main project references, both referencing the Bridge assembly.

Each should contain one file. Main should contain App.cs as follows:
```csharp
using System;
using System.Collections.Generic;
using LibraryProject;

namespace Main
{ 
    public class Program
    {
        public static void Main()
        {
            var inputTable = new System.Collections.Generic.List<System.Collections.Generic.List<string>>
            {
                new List<string>()
            };

            Console.WriteLine(inputTable.Count);
        }
    }

    public class UnusedClass
    {
        public void UnusedMethod()
        {
            ListMaker.Make();
        }
    }
}
```

LibraryProject should contain List.cs as follows:
```csharp
namespace LibraryProject
{
    internal class List<T> {}
    
    public static class ListMaker
    {
        public static void Make()
        {
            new List<string>();
        }
    }
}
```

Interestingly, it fails to replicate if you do any of the following:

- Remove UnusedClass

- Move UnusedClass into LibraryProject; testing while cutting down the replication to this tells me that it's allowed to be in a third project of its own, but if I move it into LibraryProject itself the bug goes away.

- Change
```csharp
            var inputTable = new System.Collections.Generic.List<System.Collections.Generic.List<string>>
            {
                new List<string>()
            };
```
to
```csharp
            var inputTable = new System.Collections.Generic.List<System.Collections.Generic.List<string>>
            {
                new System.Collections.Generic.List<string>()
            };
```

- Move "using System.Collections.Generic" inside the "namespace Main" block. Still broken if you move "using LibraryProject" inside the namespace block as well, though.

- Move the List class inside the ListMaker class.

### Expected Result

```js
1
```

### Actual Result

```js
0
```

**Comment by geoffreymcgill** [comment]
Hi @jbeshir – Thank you for the excellent report. We will investigate and try to fix right away.

**Comment by jbeshir** [comment]
I've gotten a copy of a solution replicating the bug uploaded at https://drive.google.com/file/d/0B_GA-etkVzZWb3dnaWVfRXBqYWs/view?usp=sharing, which might save time putting one together.

**Comment by fabriciomurta** [comment]
The associated pull request has just been merged in. The fix will be available to Bridge 16.4.0.
